lib/ssoe/service.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+require 'common/client/base'
+require 'common/client/concerns/monitoring'
+require 'ssoe/configuration'
+require 'nokogiri'
+
+module SSOe
+  class Service < Common::Client::Base
+    include Common::Client::Concerns::Monitoring
+
+    configuration SSOe::Configuration
+
+    STATSD_KEY_PREFIX = 'api.ssoe'
+
+    CONNECTION_ERRORS = [
+      Faraday::ConnectionFailed,
+      Common::Client::Errors::ClientError,
+      Common::Exceptions::GatewayTimeout,
+      Breakers::OutageException
+    ].freeze
+
+    # rubocop:disable Metrics/ParameterLists
+    def get_traits(credential_method: nil,
+                   credential_id: nil,
+                   first_name: nil,
+                   last_name: nil,
+                   birth_date: nil,
+                   ssn: nil,
+                   email: nil,
+                   phone: nil,
+                   street1: nil,
+                   city: nil,
+                   state: nil,
+                   zipcode: nil)
```

Comment:
With this amount of args being passed around in multiple places it could start to get unwieldy. It might be a good idea to extract related fields into separate classes (something like `SSOe::Models::User` and `SSOe::Models::Address`). This would greatly improve readability and maintainability. Then your `get_traits` and other methods would only pass around 4 args

```ruby
def get_traits(credential_method:, credential_id:, user:, address:)
  # ...
end
```